0.17.0 compile error on Big Sur

Unable to upgrade via Brew on Big Sur to 0.17.0:

```
$ brew upgrade arena
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 3 taps (homebrew/core, homebrew/cask and caskroom/cask).
==> Updated Formulae
Updated 1 formula.
==> Updated Casks
ibackup-viewer                                                               ibackup-viewer

Warning: You are using macOS 11.0.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this pre-release version.

==> Upgrading 1 outdated package:
finestructure/tap/arena 0.16.1 -> 0.17.0
==> Upgrading finestructure/tap/arena 0.16.1 -> 0.17.0 
==> Downloading https://github.com/finestructure/Arena/archive/0.17.0.tar.gz
==> Downloading from https://codeload.github.com/finestructure/Arena/tar.gz/0.17.0
######################################################################## 100.0%
==> swift build --configuration release --disable-sandbox
Last 15 lines from /Users/leo/Library/Logs/Homebrew/arena/01.swift:
    memcpy(buf.baseAddress!, UnsafePointer<UInt8>(bytes), buf.count)
                                                  ^~~~~
/private/tmp/arena-20200813-50476-kulb1b/Arena-0.17.0/.build/checkouts/swift-llbuild/products/llbuildSwift/Internals.swift:34:51: note: use the 'withUnsafeBufferPointer' method on Array in order to explicitly convert argument to buffer pointer valid for a defined scope
    memcpy(buf.baseAddress!, UnsafePointer<UInt8>(bytes), buf.count)
                                                  ^
[91/94] Compiling llbuildBuildSystem BuildSystem.cpp
[92/94] Compiling TSCBasic Await.swift
[93/95] Compiling TSCUtility Archiver.swift
[94/99] Compiling LLBuildManifest BuildManifest.swift
/private/tmp/arena-20200813-50476-kulb1b/Arena-0.17.0/.build/checkouts/swift-package-manager/Sources/LLBuildManifest/BuildManifest.swift:44:9: error: overlapping accesses to 'self.targets', but modification requires exclusive access; consider copying to a local variable
        targets[target, default: createTarget(target)].nodes.append(node)
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/arena-20200813-50476-kulb1b/Arena-0.17.0/.build/checkouts/swift-package-manager/Sources/LLBuildManifest/BuildManifest.swift:44:34: note: conflicting access is here
        targets[target, default: createTarget(target)].nodes.append(node)
                                 ^~~~~~~~~~~~

Do not report this issue to Homebrew/brew or Homebrew/core!


Error: You are using macOS 11.0.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this pre-release version.
```

--- a/Package.resolved
+++ b/Package.resolved
@@ -36,33 +36,6 @@
           "revision": "9f04d1ff1afbccd02279338a2c91e5f27c45e93a",
           "version": "0.0.5"
         }
-      },
-      {
-        "package": "llbuild",
-        "repositoryURL": "https://github.com/apple/swift-llbuild.git",
-        "state": {
-          "branch": "swift-5.2-branch",
-          "revision": "d6dd04202d509f91d61aef89236db0da8f27d356",
-          "version": null
-        }
-      },
-      {
-        "package": "SwiftPM",
-        "repositoryURL": "https://github.com/apple/swift-package-manager",
-        "state": {
-          "branch": "swift-5.2-RELEASE",
-          "revision": "f2318da4f96b0607ade27eff51558206f57c2f98",
-          "version": null
-        }
-      },
-      {
-        "package": "swift-tools-support-core",
-        "repositoryURL": "https://github.com/apple/swift-tools-support-core.git",
-        "state": {
-          "branch": "swift-5.2-branch",
-          "revision": "7ecf17a83eab20cbd700d7e45d66c03409bc72d0",
-          "version": null
-        }
       }
     ]
   },

--- a/Package.resolved
+++ b/Package.resolved
@@ -20,6 +20,15 @@
         }
       },
       {
+        "package": "SemanticVersion",
+        "repositoryURL": "https://github.com/SwiftPackageIndex/SemanticVersion",
+        "state": {
+          "branch": null,
+          "revision": "f751d9d68d8ec4917c951f92eb4ce0646475868c",
+          "version": "0.2.0"
+        }
+      },
+      {
         "package": "ShellOut",
         "repositoryURL": "https://github.com/JohnSundell/ShellOut.git",
         "state": {

--- a/Package.swift
+++ b/Package.swift
@@ -13,7 +13,6 @@ let package = Package(
     ],
     dependencies: [
         .package(url: "https://github.com/apple/swift-argument-parser", from: "0.0.1"),
-        .package(url: "https://github.com/apple/swift-package-manager", .revision("swift-5.2-RELEASE")),
         .package(url: "https://github.com/finestructure/Parser", from: "0.0.0"),
         .package(url: "https://github.com/JohnSundell/ShellOut.git", from: "2.0.0"),
         .package(url: "https://github.com/mxcl/Path.swift.git", from: "1.0.0"),
@@ -24,7 +23,7 @@ let package = Package(
             dependencies: ["ArenaCore"]),
         .target(
             name: "ArenaCore",
-            dependencies: ["ArgumentParser", "Parser", "Path", "ShellOut", "SwiftPM-auto"]),
+            dependencies: ["ArgumentParser", "Parser", "Path", "ShellOut"]),
         .testTarget(
             name: "ArenaTests",
             dependencies: ["ArenaCore"]),

--- a/Package.swift
+++ b/Package.swift
@@ -16,6 +16,7 @@ let package = Package(
         .package(url: "https://github.com/finestructure/Parser", from: "0.0.0"),
         .package(url: "https://github.com/JohnSundell/ShellOut.git", from: "2.0.0"),
         .package(url: "https://github.com/mxcl/Path.swift.git", from: "1.0.0"),
+        .package(url: "https://github.com/SwiftPackageIndex/SemanticVersion", from: "0.2.0")
     ],
     targets: [
         .target(
@@ -23,7 +24,7 @@ let package = Package(
             dependencies: ["ArenaCore"]),
         .target(
             name: "ArenaCore",
-            dependencies: ["ArgumentParser", "Parser", "Path", "ShellOut"]),
+            dependencies: ["ArgumentParser", "Parser", "Path", "SemanticVersion", "ShellOut"]),
         .testTarget(
             name: "ArenaTests",
             dependencies: ["ArenaCore"]),

--- a/Sources/ArenaCore/ArenaCommand.swift
+++ b/Sources/ArenaCore/ArenaCommand.swift
@@ -204,7 +204,7 @@ extension Arena {
                 zip(dependencies,
                     try dependencies.compactMap {
                         $0.path ?? $0.checkoutDir(projectDir: projectPath)
-                    }.compactMap { try getPackageInfo(for: $0) } )
+                    }.compactMap { try getPackageInfo(in: $0) } )
             )
             let libs = packageInfo.flatMap { $0.1.libraries }
             if libs.isEmpty { throw ArenaError.noLibrariesFound }

--- a/Sources/ArenaCore/Dependency.swift
+++ b/Sources/ArenaCore/Dependency.swift
@@ -7,7 +7,7 @@
 
 import ArgumentParser
 import Foundation
-import PackageModel
+//import PackageModel
 import Path
 import Parser
 

--- a/Sources/ArenaCore/Dependency.swift
+++ b/Sources/ArenaCore/Dependency.swift
@@ -16,7 +16,7 @@ public struct Dependency: Equatable, Hashable, Codable {
     public var url: URL
     public var requirement: Requirement
 
-    static let defaultRequirement: Requirement = .from("0.0.0")
+    static let defaultRequirement: Requirement = .from(.init(0, 0, 0))
 
     public init(url: URL, requirement: Requirement) {
         precondition(url.scheme != nil, "scheme must not be nil (i.e. one of https, http, file)")

--- a/Sources/ArenaCore/Extensions.swift
+++ b/Sources/ArenaCore/Extensions.swift
@@ -8,6 +8,7 @@
 import ArgumentParser
 import Foundation
 import Path
+import SemanticVersion
 import ShellOut
 
 

--- a/Sources/ArenaCore/Github.swift
+++ b/Sources/ArenaCore/Github.swift
@@ -7,7 +7,7 @@
 
 
 import Foundation
-import PackageModel
+//import PackageModel
 import Parser
 
 

--- a/Sources/ArenaCore/Github.swift
+++ b/Sources/ArenaCore/Github.swift
@@ -9,6 +9,7 @@
 import Foundation
 //import PackageModel
 import Parser
+import SemanticVersion
 
 
 let requestTimeout = 5
@@ -17,7 +18,7 @@ let requestTimeout = 5
 struct GithubClient {
     var latestRelease: (GithubRepository) -> Release?
     var tags: (GithubRepository) -> [Tag]
-    var latestVersion: (GithubRepository) -> Version?
+    var latestVersion: (GithubRepository) -> SemanticVersion?
 }
 
 
@@ -39,7 +40,7 @@ struct Release: Decodable {
         case tagName = "tag_name"
     }
 
-    var version: Version? {
+    var version: SemanticVersion? {
         Parser.version.run(tagName).result
     }
 }
@@ -115,7 +116,7 @@ func tagsRequest(for repository: GithubRepository) -> [Tag] {
 }
 
 
-func latestVersionRequest(for repository: GithubRepository) -> Version? {
+func latestVersionRequest(for repository: GithubRepository) -> SemanticVersion? {
     if
         let release = Current.githubClient.latestRelease(repository),
         let version = release.version {
@@ -123,7 +124,7 @@ func latestVersionRequest(for repository: GithubRepository) -> Version? {
     }
     let versions = Current.githubClient.tags(repository)
         .map(\.name)
-        .compactMap(Version.init(string:))
+        .compactMap(SemanticVersion.init)
         .sorted(by: >)
     if let latest = versions.first {
         return latest

--- a/Sources/ArenaCore/Github.swift
+++ b/Sources/ArenaCore/Github.swift
@@ -7,7 +7,6 @@
 
 
 import Foundation
-//import PackageModel
 import Parser
 import SemanticVersion
 

--- a/Sources/ArenaCore/Helpers.swift
+++ b/Sources/ArenaCore/Helpers.swift
@@ -6,44 +6,33 @@
 //
 
 import Foundation
-//import PackageLoading
 import Path
-//import Workspace
-
-
-// see: https://github.com/apple/swift-package-manager/blob/master/Examples/package-info/Sources/package-info/main.swift
-let swiftCompiler: AbsolutePath = {
-    let string: String
-    #if os(macOS)
-    string = try! Process.checkNonZeroExit(args: "xcrun", "--sdk", "macosx", "-f", "swiftc").spm_chomp()
-    #else
-    string = try! Process.checkNonZeroExit(args: "which", "swiftc").spm_chomp()
-    #endif
-    return AbsolutePath(string)
-}()
 
 
 public struct PackageInfo {
     var name: String
-    var path: AbsolutePath
+    var path: Path
     var libraries: [String]
 }
 
 
 public func getPackageInfo(for package: Path) throws -> PackageInfo {
-    let path = AbsolutePath(package.string)
-    let manifest = try ManifestLoader.loadManifest(packagePath: path,
-                                                   swiftCompiler: swiftCompiler,
-                                                   packageKind: .remote)
-    let libs = manifest.products.filter { p in
-        if case .library = p.type {
-            return true
-        } else {
-            return false
-        }
-    }
-    .map { $0.name }
-    return PackageInfo(name: manifest.name, path: path, libraries: libs)
+//    let path = AbsolutePath(package.string)
+//    let manifest = try ManifestLoader.loadManifest(packagePath: path,
+//                                                   swiftCompiler: swiftCompiler,
+//                                                   packageKind: .remote)
+//    let libs = manifest.products.filter { p in
+//        if case .library = p.type {
+//            return true
+//        } else {
+//            return false
+//        }
+//    }
+//    .map { $0.name }
+    let name = "" // FIXME: was manifest.name
+    let path = package  // FIXME: was AbsolutePath(package.string)
+    let libs = [String]()  // FIXME
+    return PackageInfo(name: name, path: path, libraries: libs)
 }
 
 

--- a/Sources/ArenaCore/Helpers.swift
+++ b/Sources/ArenaCore/Helpers.swift
@@ -11,28 +11,27 @@ import Path
 
 public struct PackageInfo {
     var name: String
-    var path: Path
     var libraries: [String]
 }
 
 
-public func getPackageInfo(for package: Path) throws -> PackageInfo {
-//    let path = AbsolutePath(package.string)
-//    let manifest = try ManifestLoader.loadManifest(packagePath: path,
-//                                                   swiftCompiler: swiftCompiler,
-//                                                   packageKind: .remote)
-//    let libs = manifest.products.filter { p in
-//        if case .library = p.type {
-//            return true
-//        } else {
-//            return false
-//        }
-//    }
-//    .map { $0.name }
-    let name = "" // FIXME: was manifest.name
-    let path = package  // FIXME: was AbsolutePath(package.string)
-    let libs = [String]()  // FIXME
-    return PackageInfo(name: name, path: path, libraries: libs)
+func dumpPackage(at path: Path) throws -> Manifest {
+    let json = try shellOut(to: .init(string: "swift package dump-package"), at: path)
+    return try JSONDecoder().decode(Manifest.self, from: Data(json.utf8))
+}
+
+
+public func getPackageInfo(in directory: Path) throws -> PackageInfo {
+    let manifest = try dumpPackage(at: directory)
+    let libs = manifest.products.filter { p in
+        if case .library = p.type {
+            return true
+        } else {
+            return false
+        }
+    }
+    .map { $0.name }
+    return PackageInfo(name: manifest.name, libraries: libs)
 }
 
 
@@ -53,3 +52,5 @@ func zip<A, B>(_ a: A?, _ b: B?) -> (A, B)? {
         return nil
     }
 }
+
+

--- a/Sources/ArenaCore/Helpers.swift
+++ b/Sources/ArenaCore/Helpers.swift
@@ -6,9 +6,9 @@
 //
 
 import Foundation
-import PackageLoading
+//import PackageLoading
 import Path
-import Workspace
+//import Workspace
 
 
 // see: https://github.com/apple/swift-package-manager/blob/master/Examples/package-info/Sources/package-info/main.swift

--- a/Sources/ArenaCore/Manifest.swift
+++ b/Sources/ArenaCore/Manifest.swift
@@ -1,0 +1,61 @@
+import Foundation
+
+
+// `Manifest` is mirroring what `dump-package` presumably renders into JSON
+// https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html
+// Mentioning this in particular with regards to optional values, like `platforms`
+// vs mandatory ones like `products`
+//
+//Package(
+//    name: String,
+//    platforms: [SupportedPlatform]? = nil,
+//    products: [Product] = [],
+//    dependencies: [Package.Dependency] = [],
+//    targets: [Target] = [],
+//    swiftLanguageVersions: [SwiftVersion]? = nil,
+//    cLanguageStandard: CLanguageStandard? = nil,
+//    cxxLanguageStandard: CXXLanguageStandard? = nil
+//)
+
+
+struct Manifest: Decodable, Equatable {
+    struct Platform: Decodable, Equatable {
+        enum Name: String, Decodable, Equatable, CaseIterable {
+            case macos
+            case ios
+            case tvos
+            case watchos
+        }
+        var platformName: Name
+        var version: String
+    }
+    struct Product: Decodable, Equatable {
+        enum `Type`: String, CodingKey, CaseIterable {
+            case executable
+            case library
+        }
+        var name: String
+        var type: `Type`
+    }
+    var name: String
+    var platforms: [Platform]?
+    var products: [Product]
+    var swiftLanguageVersions: [String]?
+}
+
+
+extension Manifest.Product.`Type`: Decodable {
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Self.self)
+        for k in Self.allCases {
+            if let _ = try? container.decodeNil(forKey: k) {
+                self = k
+                return
+            }
+
+        }
+        throw DecodingError.dataCorrupted(
+            DecodingError.Context(codingPath: container.codingPath,
+                                  debugDescription: "none of the required keys found"))
+    }
+}

--- a/Sources/ArenaCore/Parser+ext.swift
+++ b/Sources/ArenaCore/Parser+ext.swift
@@ -6,7 +6,7 @@
 //
 
 import Foundation
-import PackageModel
+//import PackageModel
 import Parser
 import Path
 

--- a/Sources/ArenaCore/Parser+ext.swift
+++ b/Sources/ArenaCore/Parser+ext.swift
@@ -9,6 +9,7 @@ import Foundation
 //import PackageModel
 import Parser
 import Path
+import SemanticVersion
 
 
 // https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
@@ -19,10 +20,10 @@ let AllowedEndBranchCharacters = AllowedBranchCharacters.subtracting(CharacterSe
 let AllowedRevisionCharacters = CharacterSet.whitespacesAndNewlines.inverted
 
 
-extension Parser where A == Version {
-    static var version: Parser<Version> {
+extension Parser where A == SemanticVersion {
+    static var version: Parser<SemanticVersion> {
         zip(int, literal("."), int, literal("."), int).map { major, _, minor, _, patch in
-            Version(major, minor, patch)
+            SemanticVersion(major, minor, patch)
         }
     }
 }
@@ -52,7 +53,7 @@ extension Parser where A == RefSpec {
         ]).map { _, minVersion, rangeOp, maxVersion in
             rangeOp == "..<"
                 ? .range(minVersion..<maxVersion)
-                : .range(minVersion..<Version(maxVersion.major, maxVersion.minor, maxVersion.patch + 1))
+                : .range(minVersion..<SemanticVersion(maxVersion.major, maxVersion.minor, maxVersion.patch + 1))
         }
     }
 

--- a/Sources/ArenaCore/Parser+ext.swift
+++ b/Sources/ArenaCore/Parser+ext.swift
@@ -6,7 +6,6 @@
 //
 
 import Foundation
-//import PackageModel
 import Parser
 import Path
 import SemanticVersion

--- a/Sources/ArenaCore/RefSpec.swift
+++ b/Sources/ArenaCore/RefSpec.swift
@@ -5,7 +5,6 @@
 //  Created by Sven A. Schmidt on 01/01/2020.
 //
 
-//import PackageModel
 import SemanticVersion
 
 

--- a/Sources/ArenaCore/RefSpec.swift
+++ b/Sources/ArenaCore/RefSpec.swift
@@ -5,7 +5,7 @@
 //  Created by Sven A. Schmidt on 01/01/2020.
 //
 
-import PackageModel
+//import PackageModel
 
 
 enum RefSpec: Equatable {

--- a/Sources/ArenaCore/RefSpec.swift
+++ b/Sources/ArenaCore/RefSpec.swift
@@ -6,13 +6,14 @@
 //
 
 //import PackageModel
+import SemanticVersion
 
 
 enum RefSpec: Equatable {
     case branch(String)
-    case exact(Version)
-    case from(Version)
+    case exact(SemanticVersion)
+    case from(SemanticVersion)
     case noVersion
-    case range(Range<Version>)
+    case range(Range<SemanticVersion>)
     case revision(String)
 }

--- a/Sources/ArenaCore/Requirement.swift
+++ b/Sources/ArenaCore/Requirement.swift
@@ -5,7 +5,6 @@
 //  Created by Sven A. Schmidt on 31/12/2019.
 //
 
-//import PackageModel
 import SemanticVersion
 
 

--- a/Sources/ArenaCore/Requirement.swift
+++ b/Sources/ArenaCore/Requirement.swift
@@ -6,19 +6,20 @@
 //
 
 //import PackageModel
+import SemanticVersion
 
 
 public enum Requirement: Equatable, Hashable {
-    case exact(Version)
-    case range(Range<Version>)
+    case exact(SemanticVersion)
+    case range(Range<SemanticVersion>)
     case revision(String)
     case branch(String)
     case path
-    case from(Version)
+    case from(SemanticVersion)
     case noVersion
 
-    public static func upToNextMajor(from version: Version) -> Requirement {
-        return .range(version..<Version(version.major + 1, 0, 0))
+    public static func upToNextMajor(from version: SemanticVersion) -> Requirement {
+        return .range(version..<SemanticVersion(version.major + 1, 0, 0))
     }
 }
 
@@ -27,12 +28,12 @@ extension Requirement: Codable {
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
 
-        if let v = try? container.decode(Version.self, forKey: .exact) {
+        if let v = try? container.decode(SemanticVersion.self, forKey: .exact) {
             self = .exact(v)
             return
         }
 
-        if let v = try? container.decode(Range<Version>.self, forKey: .range) {
+        if let v = try? container.decode(Range<SemanticVersion>.self, forKey: .range) {
             self = .range(v)
             return
         }
@@ -52,12 +53,12 @@ extension Requirement: Codable {
             return
         }
 
-        if let v = try? container.decode(Version.self, forKey: .from) {
+        if let v = try? container.decode(SemanticVersion.self, forKey: .from) {
             self = .from(v)
             return
         }
 
-        if let _ = try? container.decode(Version.self, forKey: .noVersion) {
+        if let _ = try? container.decode(SemanticVersion.self, forKey: .noVersion) {
             self = .noVersion
             return
         }

--- a/Sources/ArenaCore/Requirement.swift
+++ b/Sources/ArenaCore/Requirement.swift
@@ -5,7 +5,7 @@
 //  Created by Sven A. Schmidt on 31/12/2019.
 //
 
-import PackageModel
+//import PackageModel
 
 
 public enum Requirement: Equatable, Hashable {

--- a/Sources/ArenaCore/String+ext.swift
+++ b/Sources/ArenaCore/String+ext.swift
@@ -7,6 +7,7 @@
 
 //import PackageModel
 import Parser
+import SemanticVersion
 
 
 extension String {
@@ -14,7 +15,7 @@ extension String {
         Parser.dependency.run(self).result
     }
 
-    public var version: Version? {
+    public var version: SemanticVersion? {
         Parser.version.run(self).result
     }
 }

--- a/Sources/ArenaCore/String+ext.swift
+++ b/Sources/ArenaCore/String+ext.swift
@@ -5,7 +5,7 @@
 //  Created by Sven A. Schmidt on 13/03/2020.
 //
 
-import PackageModel
+//import PackageModel
 import Parser
 
 

--- a/Sources/ArenaCore/String+ext.swift
+++ b/Sources/ArenaCore/String+ext.swift
@@ -5,7 +5,6 @@
 //  Created by Sven A. Schmidt on 13/03/2020.
 //
 
-//import PackageModel
 import Parser
 import SemanticVersion
 

--- a/Tests/ArenaTests/ArenaTests.swift
+++ b/Tests/ArenaTests/ArenaTests.swift
@@ -1,7 +1,7 @@
 @testable import ArenaCore
 import Path
 import Parser
-import Workspace
+import SemanticVersion
 import XCTest
 
 
@@ -16,7 +16,7 @@ extension GithubClient {
     static let mock = Self(
         latestRelease: { _ in Release(tagName: "1.2.3") },
         tags: { _ in [Tag(name: "1.2.3")] },
-        latestVersion: { _ in "1.2.3" }
+        latestVersion: { _ in SemanticVersion("1.2.3") }
     )
 }
 
@@ -33,17 +33,17 @@ final class ArenaTests: XCTestCase {
         Current = .mock
     }
 
-    func test_loadManifest() throws {
-        let p = checkoutsDirectory/"swift-package-manager"
-        print(p)
-        let package = AbsolutePath(p.string)
-        let manifest = try ManifestLoader.loadManifest(packagePath: package,
-                                                       swiftCompiler: swiftCompiler,
-                                                       packageKind: .remote)
-        XCTAssertEqual(manifest.name, "SwiftPM")
-        XCTAssertEqual(manifest.products.map { $0.name }, ["SwiftPM", "SwiftPM-auto", "PackageDescription"])
-        XCTAssertEqual(manifest.products.map { $0.type }, [.library(.dynamic), .library(.automatic), .library(.dynamic)])
-    }
+//    func test_loadManifest() throws {
+//        let p = checkoutsDirectory/"swift-package-manager"
+//        print(p)
+//        let package = AbsolutePath(p.string)
+//        let manifest = try ManifestLoader.loadManifest(packagePath: package,
+//                                                       swiftCompiler: swiftCompiler,
+//                                                       packageKind: .remote)
+//        XCTAssertEqual(manifest.name, "SwiftPM")
+//        XCTAssertEqual(manifest.products.map { $0.name }, ["SwiftPM", "SwiftPM-auto", "PackageDescription"])
+//        XCTAssertEqual(manifest.products.map { $0.type }, [.library(.dynamic), .library(.automatic), .library(.dynamic)])
+//    }
 
     func test_getPackageInfo() throws {
         let package = checkoutsDirectory/"swift-package-manager"

--- a/Tests/ArenaTests/ArenaTests.swift
+++ b/Tests/ArenaTests/ArenaTests.swift
@@ -33,22 +33,9 @@ final class ArenaTests: XCTestCase {
         Current = .mock
     }
 
-//    func test_loadManifest() throws {
-//        let p = checkoutsDirectory/"swift-package-manager"
-//        print(p)
-//        let package = AbsolutePath(p.string)
-//        let manifest = try ManifestLoader.loadManifest(packagePath: package,
-//                                                       swiftCompiler: swiftCompiler,
-//                                                       packageKind: .remote)
-//        XCTAssertEqual(manifest.name, "SwiftPM")
-//        XCTAssertEqual(manifest.products.map { $0.name }, ["SwiftPM", "SwiftPM-auto", "PackageDescription"])
-//        XCTAssertEqual(manifest.products.map { $0.type }, [.library(.dynamic), .library(.automatic), .library(.dynamic)])
-//    }
-
     func test_getPackageInfo() throws {
-        let package = checkoutsDirectory/"swift-package-manager"
-        XCTAssertEqual(try getPackageInfo(for: package).libraries,
-                       ["SwiftPM", "SwiftPM-auto", "PackageDescription"])
+        let dir = checkoutsDirectory/"Parser"
+        XCTAssertEqual(try getPackageInfo(in: dir).libraries, ["Parser"])
     }
 
     func test_args_multiple_deps() throws {

--- a/Tests/ArenaTests/Helpers.swift
+++ b/Tests/ArenaTests/Helpers.swift
@@ -1,0 +1,8 @@
+import SemanticVersion
+
+
+extension SemanticVersion: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self = SemanticVersion(value)!
+    }
+}

--- a/Tests/ArenaTests/LocalIntegrationTests.swift
+++ b/Tests/ArenaTests/LocalIntegrationTests.swift
@@ -11,11 +11,12 @@ import XCTest
 
 class LocalIntegrationTests: XCTestCase {
     override func setUpWithError() throws {
-//        try XCTSkipUnless(ProcessInfo().hostName == "luna.local", "fails on CI, only run locally")
         Current = .live
     }
 
     func test_ouput() throws {
+        try XCTSkipUnless(ProcessInfo().hostName == "luna.local", "skip on CI due to '../../tmp' path prefix in output, only run locally")
+
         let output = OutputListener()
         output.openConsolePipe()
 

--- a/Tests/ArenaTests/LocalIntegrationTests.swift
+++ b/Tests/ArenaTests/LocalIntegrationTests.swift
@@ -11,7 +11,7 @@ import XCTest
 
 class LocalIntegrationTests: XCTestCase {
     override func setUpWithError() throws {
-        try XCTSkipUnless(ProcessInfo().hostName == "luna.local", "fails on CI, only run locally")
+//        try XCTSkipUnless(ProcessInfo().hostName == "luna.local", "fails on CI, only run locally")
         Current = .live
     }
 
@@ -30,7 +30,6 @@ class LocalIntegrationTests: XCTestCase {
                 ➡️  Package: https://github.com/finestructure/ArenaTest @ exact(0.0.3)
                 🔧 Resolving package dependencies ...
                 📔 Libraries found: ArenaTest
-                🔨 Building package dependencies ...
                 ✅ Created project in folder 'ArenaIntegrationTest'
                 Run
                   open ArenaIntegrationTest/ArenaIntegrationTest.xcworkspace

--- a/Tests/ArenaTests/ParserTests.swift
+++ b/Tests/ArenaTests/ParserTests.swift
@@ -8,7 +8,7 @@
 @testable import ArenaCore
 import Path
 import Parser
-import Workspace
+import SemanticVersion
 import XCTest
 
 
@@ -18,8 +18,8 @@ class ParserTests: XCTestCase {
     }
 
     func test_parse_version() throws {
-        XCTAssertEqual(Parser.version.run("1.2.3"), Match(result: Version(1, 2, 3), rest: ""))
-        XCTAssertEqual(Parser.version.run("1.2.3="), Match(result: Version(1, 2, 3), rest: "="))
+        XCTAssertEqual(Parser.version.run("1.2.3"), Match(result: SemanticVersion(1, 2, 3), rest: ""))
+        XCTAssertEqual(Parser.version.run("1.2.3="), Match(result: SemanticVersion(1, 2, 3), rest: "="))
     }
 
     func test_parse_requirement() throws {

